SearchManager.cls

```diff
@@ -0,0 +1,37 @@
+public class SearchManager {
+    String searchText;
+    String subject;
+    String body;
+    List<Contact> contacts;
+	
+    public SearchManager(String searchText, String subject, String body, object[] newTrigger) {
+        this.searchText = searchText;
+        this.subject = subject;
+        this.body = body;
+      	this.contacts = (List<Contact>) newTrigger;
+    }
+    
+    public Boolean searchContact(){
+        String contactName = '';
+        List<String> contactsList = new List<String>();
+        Id userId = UserInfo.getUserId();
+        User currentUser =[SELECT Email FROM User WHERE Id=:userId];
+        String addresUser = String.valueOf(currentUser.Email);
+
+        for(Contact contact : contacts){
+            ID acctID = contact.AccountId;
+            Account account = [SELECT Name FROM Account WHERE Id=:acctID];
```

Comment:
@lubomyrV 
Do not ever use SOQL query or DML operation in loops!!!
You can hit SF governor limits ( check this link https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_gov_limits.htm )